py/desispec/tile_qa.py

```diff
@@ -0,0 +1,221 @@
+    tsnr2_key="TSNR2_LRG"
+    if tsnr2_key in scores.dtype.names :
+        tile_fiberqa_table[tsnr2_key] = np.zeros(targetids.size)
+        for i,tid in enumerate(targetids) :
+            if tid in scores_tid_to_index :
+                tile_fiberqa_table[tsnr2_key][i] = scores[tsnr2_key][scores_tid_to_index[tid]]
+
+    # add ZBEST info
+    zbest_tid_to_index = {tid:index for index,tid in enumerate(zbests["TARGETID"])}
+    keys=["Z","SPECTYPE","DELTACHI2"]
+    for k in keys :
+        tile_fiberqa_table[k] = np.zeros(targetids.size,dtype=zbests[k].dtype)
+    zbest_ii=[]
+    fiberqa_ii=[]
+    for i,tid in enumerate(targetids) :
+        if tid in zbest_tid_to_index :
+            zbest_ii.append(zbest_tid_to_index[tid])
+            fiberqa_ii.append(i)
+    for k in keys :
+        tile_fiberqa_table[k][fiberqa_ii] = zbests[k][zbest_ii]
+
+
+
+    # QAFIBERSTATUS is OR of input exposures
+    tile_fiberqa_table["QAFIBERSTATUS"]=np.zeros(targetids.size,dtype=exposure_fiberqa_tables["QAFIBERSTATUS"].dtype)
+    for i,tid in enumerate(targetids) :
+        jj = (exposure_fiberqa_tables["TARGETID"]==tid)
+        tile_fiberqa_table["QAFIBERSTATUS"][i] = np.bitwise_or.reduce(exposure_fiberqa_tables['QAFIBERSTATUS'][jj])
+
+    # EFFTIME
+    tile_fiberqa_table["EFFTIME_SPEC"]=np.zeros(targetids.size,dtype=exposure_fiberqa_tables["EFFTIME_SPEC"].dtype)
+    for i,tid in enumerate(targetids) :
+        jj = (exposure_fiberqa_tables["TARGETID"]==tid)
+        tile_fiberqa_table["EFFTIME_SPEC"][i] = np.sum(exposure_fiberqa_tables['EFFTIME_SPEC'][jj])
+
+
+    qa_params=get_qa_params()["exposure_qa"]
+    bad_fibers_mask=fibermask.mask(qa_params["bad_qafstatus_mask"])
+
+    good_fibers = np.where((tile_fiberqa_table['QAFIBERSTATUS']&bad_fibers_mask)==0)[0]
+    good_petals = np.unique(tile_fiberqa_table['PETAL_LOC'][good_fibers])
+
+    npetal=10
+    tile_petalqa_table = Table()
+    petals=np.unique(exposure_fiberqa_tables["PETAL_LOC"])
+    tile_petalqa_table["PETAL_LOC"]=np.arange(npetal,dtype=int)
+    keys=['WORSTREADNOISE', 'NGOODPOS', 'NSTDSTAR', 'STARRMS', 'TSNR2FRA', 'NCFRAME',\
+          'BSKYTHRURMS', 'BSKYCHI2PDF', 'RSKYTHRURMS', 'RSKYCHI2PDF', 'ZSKYTHRURMS', 'ZSKYCHI2PDF',\
+          'BTHRUFRAC', 'RTHRUFRAC', 'ZTHRUFRAC']
+    for k in keys :
+        tile_petalqa_table[k]=np.zeros(npetal)
+    for petal in petals :
+        ii=(exposure_petalqa_tables["PETAL_LOC"]==petal)
+        for k in keys :
+            tile_petalqa_table[k][petal]=np.mean(exposure_petalqa_tables[k][ii])
+
+    # EFFTIME
+    tile_petalqa_table["EFFTIME_SPEC"]=np.zeros(npetal)
+    for petal in petals :
+        entries=(tile_fiberqa_table['PETAL_LOC'] == petal)
+        tile_petalqa_table['EFFTIME_SPEC'][petal]=np.median(tile_fiberqa_table["EFFTIME_SPEC"][entries])
+
+    # add meta info
+    tile_fiberqa_table.meta["TILEID"]=tileid
+    tile_fiberqa_table.meta["NIGHT"]=night
```

Comment:
Since tile QA could span multiple nights, "NIGHT" seems a misnomer.  Consider "LASTNITE" (and see next comment about why not "LASTNIGHT").